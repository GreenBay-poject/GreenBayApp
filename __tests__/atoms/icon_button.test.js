import React from 'react';
import renderer from 'react-test-renderer';

import CustomIconButton from '../../atoms/icon_button'

describe('<CustomIconButton />', () => {
    it('Sample CustomIconButton', () => {
        const tree = renderer.create(<CustomIconButton ant_design_icon_name={"book"} text={"Sample"} navigation={null} page_to_go={"Dashboard"} />).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree.children.length).toBe(1);
        expect(tree.children[0].children.length).toBe(2);

    });
});