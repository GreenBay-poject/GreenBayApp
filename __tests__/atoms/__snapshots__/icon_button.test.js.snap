// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomIconButton /> Sample CustomIconButton 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#d7dbd8",
      "borderColor": "#153d37",
      "borderRadius": 20,
      "borderWidth": 0,
      "elevation": 2,
      "flex": 1,
      "marginHorizontal": 5,
      "padding": 0,
      "width": 130,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#d7dbd8",
        "borderColor": "#153d37",
        "borderRadius": 20,
        "borderWidth": 1,
        "elevation": 2,
        "flex": 1,
        "marginHorizontal": 5,
        "padding": 10,
        "width": 130,
      }
    }
  >
    <Text />
    <Text
      style={
        Object {
          "color": "#153d37",
          "fontSize": 17,
          "fontWeight": "bold",
        }
      }
    >
      Sample
    </Text>
  </View>
</View>
`;
